mypy/messages.py

```diff
@@ -1775,9 +1775,12 @@ def need_annotation_for_var(
                     recommended_type = f"Optional[{type_dec}]"
             elif node.type.type.fullname in reverse_builtin_aliases:
                 # partial types other than partial None
-                alias = reverse_builtin_aliases[node.type.type.fullname]
+                if mypy.options._based:
+                    alias = node.type.type.fullname
+                else:
+                    alias = reverse_builtin_aliases[node.type.type.fullname]
                 alias = alias.split(".")[-1]
-                if alias == "Dict":
+                if alias.lower() == "dict":
```

Comment:
what if someone makes a completely unrelated type called `DICT`

Reply:
take it up with `mypy`

Reply:
it checks for fullname earlier